sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -64,10 +64,26 @@ public Response Response
         /// </summary>
         public CancellationToken CancellationToken { get; internal set; }
 
+        private ResponseClassifier _responseClassifier;
         /// <summary>
         /// The <see cref="ResponseClassifier"/> instance to use for response classification during pipeline invocation.
         /// </summary>
-        public ResponseClassifier ResponseClassifier { get; set; }
+        public ResponseClassifier ResponseClassifier
+        {
+            get
+            {
+                return _responseClassifier;
+            }
+            set
+            {
+                _responseClassifier = value;
+
+                if (TryCustomizeClassifier(out var classifier))
```

Comment:
Right now, the only place we're calling the setter is in the DPG CreateXxRequest() methods, which don't have access to internal properties on RequestContext or HttpMessage.  This was my attempt at keeping additional public APIs off these types.  I'm happy to try a different approach - is there one you'd suggest?

Reply:
Yeah, I can see how we would need more public APis, but I think it's fine. Could you show what public APIs we would need? My mental model is that we would need one public method that applies RequestContext to the message. But I might be wrong.

Reply:
Looking at this in the next commit.

Reply:
Yeah, I was leaning away from that, because we were trying to do that in 

```
var message = _pipeline.CreateMessage(context);
```

... but since we set the classifier after the context is applied, we have to apply it again to get the classifier customizations.   In retrospect, we probably shouldn't have added the CreateMessage overload taking a context, in favor of doing the "apply" in a separate method.  Live and learn!

We basically want something along these lines:
```
            if (!context.HasClassifier)
            {
                message.ResponseClassifier = ResponseClassifier200404.Instance;
            }
            else
            {
                message.ResponseClassifier = ResponseClassifier200404.Instance.Clone();
                context.Apply(message);
            }
```

We need to know whether we have classifier customizations in order to know whether we can use the static shared classifier, or if we need to create a separate instance of it.  And then, we only want to apply classifications if we're not using the static shared instance.

Which is why I was thinking we should pass in a StatusCodeClassifier and not the whole message, but the trouble there is that StatusCodeClassifier is internal, so can't be a parameter in a public method on RequestContext.

Reply:
> We basically want something along these lines:

Yes! :-)

> but the trouble there is that StatusCodeClassifier is internal

Could we do something like the following:
```csharp
if (!context.HasClassifier) {
    message.ResponseClassifier = ResponseClassifier200404.Instance;
}
else {
    message.ResponseClassifier = StatusCodeClassifier.ResponseClassifier200404.Instance.Apply(context);
}
```
and if we moved if statement into Apply and the callers would do:
```csharp
    message.ResponseClassifier = StatusCodeClassifier.ResponseClassifier200404.Instance.Apply(context);
```

Reply:
I don't see it yet, but help me understand:

> `ResponseClassifier200404.Instance.Apply(context);`

Would this also clone the instance for the caller?

We would also need access to the added classifier information on the context in StatusCodeClassifier, so I think those would need to be public too to accomplish this.

Reply:
Yes, it would clone, if context has anything interesting to apply, otherwise it would return the shared instance.

We could do it by either exposing the APIs on context (as you observed), or exposing Apply method on the ResponseClassifier. The default implementation would always chain (using an internal ChainingClassifier type).

Reply:
Moving this into message.ApplyRequestContext().  Needed to make CoreResponseClassifier public to accomplish this.